### DISCUSSION TITLE
Warning .env not exist on artisan test

### DISCUSSION THREAD

**Discussion by luca-rigutti** [laravel/framework#49870] [Q&A]
### Laravel Version

10.42.0

### PHP Version

8.3.2

### Description

Hi,

I create a fresh new installation and start to make some test, but I have all warning for integration tests.

The error is the following: `file_get_contents(/var/www/.env): Failed to open stream: No such file or directory`

I don't have an env file because I'm building this inside a container of docker, so the env is setup on the docker-compose.
I didn't find this problem in some old versions, I don't know if is something I setup or is this the version that have this problem.

Thank you.

### Steps To Reproduce

- Create a simple Dockerfile
- Use docker-compose for keep it run with env list inside it
- Go inside and make a test
- It return the error

**Comment by gludovatza** [comment]
I get the same warning message in Azure App service container.
I know, that the **.env** file isn't required in the docker container, but if you create the .env file (with this command in the project root folder: **cp .env.example .env**), after that, you won't get the warning message, and the test will run correctly.

**Comment by kishoth-y** [comment]
I am having the same issue.
Was wondering why this happens and do we have a proper solution for this?

**Comment by uob-hillmu** [comment]
Just hit this one too, on Laravel 11.  I've solved by simply having my test script run `touch .env` inside the container before it runs `php artisan test`.  Much the same as @gludovatza's solution, except it appears to work even if the file is empty 🤷‍♂️.  This feels like a bug to me.

**Comment by samliu999** [comment]
### Root Cause:
When running phpunit tests in GitHub CI, especially feature tests, I encountered an issue:
```
cp ./.env.testing.example ./.env.testing
php artisan test --env=testing
```
This produced warnings like:
```
WARN  Tests\Feature\V1\XXXXXControllerTest
! xxxxx returns successful response → file_get_contents(/app/.env):… 0.13s
```
(Running in Docker, paths start with /app/)
Initially, this didn't happen locally. But when I removed the local `.env` file, the same warning appeared.

### Solution Attempt:
Following @uob-hillmu uob-hillmu suggestion, I added this to my CI pipeline:
```
cp ./.env.testing.example ./.env.testing
touch .env
```
This fixed the warnings.

### Investigation:
I was confused - I specified `--env=testing`, so why was Laravel looking for `.env`?
I added debug output to:
`vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/LoadEnvironmentVariables.php@setEnvironmentFilePath`


```php
echo $app->environmentPath().'/'.$file. PHP_EOL;
```
Output showed: `/var/www/html/.env.local`

**The issue**: My `.env.testing.example` and `.env.testing` had `APP_ENV=local` instead of `APP_ENV=testing`. Laravel reads the `APP_ENV` variable to determine which env file to load.

### Fix:
Changed `APP_ENV=local` to `APP_ENV=testing` in `.env.testing`. Now no warnings in CI or locally, and no need for touch .env.

***I can't guarantee my understanding is 100% correct, but this solution worked for me.
Hope this helps anyone encountering the same issue.***
